scipy/ndimage/io.py

```diff
@@ -1,49 +1,31 @@
 from __future__ import division, print_function, absolute_import
 
+
+_have_pil = True
+try:
+    from scipy.misc.pilutil import imread as _imread
+except ImportError:
+    _have_pil = False
+
+
 __all__ = ['imread']
 
-from numpy import array
 
+# Use the implementation of `imread` in `scipy.misc.pilutil.imread`.
+# If it weren't for the different names of the first arguments of
+# ndimage.io.imread and misc.pilutil.imread, we could simplify this file
+# by writing
+#     from scipy.misc.pilutil import imread
+# Unfortunately, because the argument names are different, that
+# introduces a backwards incompatibility.
 
 def imread(fname, flatten=False, mode=None):
-    """
-    Read an image from a file as an array.
-
-    Parameters
-    ----------
-    fname : str
-        Image file name, e.g. ``test.jpg``, or a file object.
-    flatten : bool, optional
-        If true, convert the output to grey-scale. Default is False.
-    mode : str, optional
-        mode to convert image to, e.g. ``RGB``.
-
-
-    Returns
-    -------
-    img_array : ndarray
-        The different colour bands/channels are stored in the
-        third dimension, such that a grey-image is MxN, an
-        RGB-image MxNx3 and an RGBA-image MxNx4.
-
-    Raises
-    ------
-    ImportError
-        If the Python Imaging Library (PIL) can not be imported.
-
-    """
-    try:
-        from PIL import Image
-    except ImportError:
-        raise ImportError("Could not import the Python Imaging Library (PIL)"
-                          " required to load image files.  Please refer to"
-                          " http://pypi.python.org/pypi/PIL/ for installation"
-                          " instructions.")
-
-    im = Image.open(fname)
-    if mode:
-        im = im.convert(mode)
-    if flatten:
-        im = im.convert('F')
-    result = array(im)
-    return result
+    if _have_pil:
+        return _imread(fname, flatten, mode)
+    raise ImportError("Could not import the Python Imaging Library (PIL)"
+                      " required to load image files.  Please refer to"
+                      " http://pypi.python.org/pypi/PIL/ for installation"
+                      " instructions.")
+
+if _have_pil:
+    imread.__doc__ = _imread.__doc__.replace('name : str', 'fname : str')
```

Comment:
This should be behind ``if imread.__doc__ is not None:``. That'll fix the TravisCI failure on the Python 2.6 build. Reason: that uses ``python -OO`` which strips docstrings.